pkg/apiserver/rest/dualwriter_mode1.go

```diff
@@ -37,3 +39,13 @@ func (d *DualWriterMode1) Create(ctx context.Context, obj runtime.Object, create
 func (d *DualWriterMode1) Get(ctx context.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
 	return d.Legacy.Get(ctx, name, options)
 }
+
+// List overrides the behavior of the generic DualWriter and reads only from LegacyStorage.
+func (d *DualWriterMode1) List(ctx context.Context, options *metainternalversion.ListOptions) (runtime.Object, error) {
+	legacy, ok := d.Legacy.(rest.Lister)
+	if !ok {
+		return nil, fmt.Errorf("legacy storage rest.Lister is missing")
```

Comment:
I think we can already use the errors package, other methods follow that pattern in main.

Reply:
Thanks for the feedback! Refactored errors here and looks better to me: [1a2bee5](https://github.com/grafana/grafana/pull/85814/commits/1a2bee5fb25508720e2a436f3d011e43520cb10e)